cpp/models/abm/infection.cpp

```diff
@@ -41,48 +69,72 @@ double mio::abm::Infection::get_viral_load(const TimePoint t) const
     }
 }
 
-// double mio::abm::ViralLoad::get_viral_load(const TimePoint t, const TimePoint start_date) const
-// {
-//     if (t <= m_peak / m_incline + start_date) {
-//         return m_incline * (t - start_date);
-//     }
-//     else {
-//         return m_peak + m_decline * (t - m_peak / m_incline - start_date);
-//     }
-// }
-// void mio::abm::ViralLoad::draw_viral_load()
-// {
-//     // These numbers are subject to change, They are going to be based on distributions backed from data.
-//     m_peak    = 5.0;
-//     m_incline = 1.0;
-//     m_decline = -1.0;
-// }
+double Infection::get_infectivity(const TimePoint& t) const
+{
+    return 1 / (1 + exp(-(m_log_norm_alpha + m_log_norm_beta * get_viral_load(t))));
+}
 
-// InfectionState mio::abm::Infection::get_infection_state(const TimePoint t) const
-// {
-//     return (*std::prev(std::lower_bound(m_infection_course.begin(), m_infection_course.end(), t))).second;
-// }
+std::shared_ptr<Virus> Infection::get_virus_type() const
+{
+    return m_virus;
+}
+
+InfectionState Infection::get_start_infection_state() const
+{
+    return m_infection_course[0].second;
+}
+
+InfectionState Infection::get_infection_state(const TimePoint& t) const
+{
+    return (*std::prev(std::lower_bound(m_infection_course.begin(), m_infection_course.end(), t, [](std::pair<TimePoint, InfectionState> state, const TimePoint& t) { return state.first <= t; } ))).second;
+}
+
+void Infection::set_detected() {
+    m_detected = true;
+}
 
-// void mio::abm::Infection::draw_infection_course(InfectionState start_state)
-// {
-//     m_infection_course.push_back(std::pair<TimePoint, InfectionState>(m_start_date, start_state));
-//     while ((m_infection_course.back().second != InfectionState::Recovered_Infected ||
-//             m_infection_course.back().second != InfectionState::Recovered_Carrier ||
-//             m_infection_course.back().second != InfectionState::Dead)) {
-//         switch (m_infection_course.back().second) {
-//         case InfectionState::Susceptible:
+bool Infection::is_detected() const {
+    return m_detected;
+}
+
+void Infection::draw_infection_course(const InfectionState& start_state)
+{
+    auto t = m_start_date;
+    m_infection_course.push_back(std::pair<TimePoint, InfectionState>(t, start_state));
+    while ((m_infection_course.back().second != InfectionState::Recovered_Infected ||
+            m_infection_course.back().second != InfectionState::Recovered_Carrier ||
+            m_infection_course.back().second != InfectionState::Dead)) {
+        switch (m_infection_course.back().second) {
+            case InfectionState::Exposed:
+                // roll out how long until carrier
```

Comment:
Yes. But wouldnt it be even recursive? I.e. determine until recovery or dead or is the idea only to draw the next step? Then we should rename the function.

Reply:
The idea is to draw all steps. This can be implemented either recursively or through a loop that checks the last entry. Maybe I am missing something.

Reply:
The idea is to draw all steps. This can be implemented either recursively or through a loop that checks the last entry. Maybe I am missing something.